hummingbot/connector/exchange/btc_markets/btc_markets_exchange.py

```diff
@@ -0,0 +1,966 @@
+                await asyncio.sleep(0.5)
+
+    async def _update_balances(self):
+        """
+        Calls REST API to update total and available balances.
+        """
+        local_asset_names = set(self._account_balances.keys())
+        remote_asset_names = set()
+        account_info = await self._api_request("GET", f"{Constants.ACCOUNTS_URL}/me/balances", {}, True)
+        for account in account_info:
+            asset_name = account["assetName"]
+            self._account_available_balances[asset_name] = Decimal(str(account["available"]))
+            self._account_balances[asset_name] = Decimal(str(account["balance"]))
+            # self._account_locked_balances[asset_name] = Decimal(str(account["locked"]))
+            remote_asset_names.add(asset_name)
+
+        asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+        for asset_name in asset_names_to_remove:
+            del self._account_available_balances[asset_name]
+            del self._account_balances[asset_name]
+
+        # as self._real_time_balance_update = false
+        # take `in_flight_orders` snapshot during `_update_balances` (see #2224)
+        self._in_flight_orders_snapshot = {k: copy.copy(v) for k, v in self._in_flight_orders.items()}
+        self._in_flight_orders_snapshot_timestamp = btc_markets_utils.get_ms_timestamp()
+
+    async def _update_order_status(self):
+        """
+        Calls REST API to get status update for each in-flight order.
+        """
+
+        last_tick = int(self._last_poll_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+        current_tick = int(self.current_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+
+        if current_tick > last_tick and len(self._in_flight_orders) > 0:
+            tracked_orders = list(self._in_flight_orders.values())
+            tasks = []
+            for tracked_order in tracked_orders:
+                order_id = await tracked_order.get_exchange_order_id()
+                tasks.append(self._api_request("GET",
+                                               f"{Constants.ORDERS_URL}/{order_id}",
+                                               {},
+                                               # Constants.ORDERS_URL,
+                                               # {"id": order_id},
+                                               True))
+            self.logger().debug(f"Polling for order status updates of {len(tasks)} orders.")
+            responses = await safe_gather(*tasks, return_exceptions=True)
+            for response in responses:
+                if isinstance(response, Exception):
+                    raise response
+                if "status" not in response:
+                    self.logger().info(f"_update_order_status result not in resp: {response}")
+                    continue
+
+                if "trades" in response:
+                    await self._process_trade_message(response)
+
```

Comment:
According to the exchange documentation (https://docs.btcmarkets.net/doc/#operation/getOrderById) the endpoint that returns the order information does not include the trades in the response. Trades should be retrieved by using the proper endpoint.